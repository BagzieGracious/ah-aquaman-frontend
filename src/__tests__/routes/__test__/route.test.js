import { shallow } from "enzyme";
import React from "react";
import Route from "../../../routes/index";

describe("Route component", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<Route />);
    expect(wrapper).toMatchSnapshot();
  });
});
