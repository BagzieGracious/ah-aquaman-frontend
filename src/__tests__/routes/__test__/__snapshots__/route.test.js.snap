// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter
        className="container"
      >
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/login"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/reset_password"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/reset_password/:token"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/logout"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/signup"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/profile"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/editprofile"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/article/:slug"
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/create_article"
          strict={true}
        />
        <ProtectedRoute
          component={[Function]}
          exact={true}
          path="/editor/:slug"
          strict={true}
        />
      </BrowserRouter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/reset_password"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/reset_password/:token"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/logout"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/signup"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/profile"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/editprofile"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/article/:slug"
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/create_article"
            strict={true}
          />,
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/editor/:slug"
            strict={true}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/reset_password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/reset_password/:token",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/logout",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/editprofile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/article/:slug",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/create_article",
            "strict": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/editor/:slug",
            "strict": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter
          className="container"
        >
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/reset_password"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/reset_password/:token"
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/logout"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/signup"
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/profile"
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/editprofile"
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/article/:slug"
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/create_article"
            strict={true}
          />
          <ProtectedRoute
            component={[Function]}
            exact={true}
            path="/editor/:slug"
            strict={true}
          />
        </BrowserRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/reset_password"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/reset_password/:token"
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/logout"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/signup"
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/profile"
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/editprofile"
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/article/:slug"
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/create_article"
              strict={true}
            />,
            <ProtectedRoute
              component={[Function]}
              exact={true}
              path="/editor/:slug"
              strict={true}
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/reset_password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/reset_password/:token",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/logout",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/profile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/editprofile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/article/:slug",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/create_article",
              "strict": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/editor/:slug",
              "strict": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
